### PR TITLE
Prevent grpc crash on bad version

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Refactored creation of span event aggregator to prevent crash of grpc when running on invalid node.js version

## Details
I also removed the flappy mongo db version tests as its dependent on having a consistent env for timings.  I also added more tests for the new fallback scenario of the span event aggregator

### CODE REVIEW

**Comment by carlo-808** [comment]
I think you need to remove `cursor-duration.tap.js` from the /versioned/mongodb/package.json files lists to clear the breaking versioned tests here.

**Comment by bizob2828** [comment]
> I think you need to remove `cursor-duration.tap.js` from the /versioned/mongodb/package.json files lists to clear the breaking versioned tests here.

bah. i was wondering what was up, thanks for the heads up!

File: lib/spans/create-span-event-aggregator.js [hunk 1]
```diff
@@ -22,14 +20,26 @@ function createSpanEventAggregator(config, collector, metrics) {
     }
     trace_observer.port = trace_observer.port.trim()
 
-    return createStreamingAggregator(config, collector, metrics)
+    try {
```
Comment (carlo-808) [inline lib/spans/create-span-event-aggregator.js:23]: Looks like a decent safety check here with relevant information logged. 👍

File: test/smoke/index/index-bad-version-streaming.tap.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
```
Comment (carlo-808) [inline test/smoke/index/index-bad-version-streaming.tap.js:8]: Nice work adding the smoke test.

**Review by carlo-808 — APPROVED** [review]
👍

**Pull request merged**